### PR TITLE
Feature/144 housing available read endpoint

### PR DESCRIPTION
**Pull request by VitaRox**
### Works!  Navigate to from the sidebar Housing Available link
-- displays all stored Housing Available listings in attractive boxes.
-- **later**: work on styles for housing-available, make more neutrally colored, link in to Property, Review, and Landlord listings.

### CODE REVIEW

File: src/app/_services/api.service.ts [hunk 1]
```diff
@@ -17,6 +18,7 @@ export class ApiService {
   public prevPage: string = "";
   public nextPage: string = "";
   public lastPage: string = "";
+  getHousingAvailable: any;
```
Comment (zaclittleberry) [inline src/app/_services/api.service.ts:21]: this doesn't do anything. please remove

File: src/app/_services/api.service.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { HttpClient, HttpParams } from '@angular/common/http';
 
 import { Property } from '_models/property.model';
 import { Landlord } from '_models/landlord.model';
+// import { HousingAvailable } from '_models/housing-available.model';
```
Comment (zaclittleberry) [inline src/app/_services/api.service.ts:6]: were you unable to get the model to work in the api service, or unsure if/how you should? on your httpClient call in your getHousingAvailables method, you can replace the cast response from <any> to <HousingAvailable[]> and it should cast the api response as an array of HousingAvailable models.

File: src/app/_pages/housing-availables/housing-availables.page.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+@Component({
+  selector: 'housing-availables-page', // should be also "-page"?
+  templateUrl: 'housing-availables.page.html',
+  styleUrls: ['housing-availables.page.scss'],
+})
+export class HousingAvailablesPage {
+
+  loading: boolean = true;
+  housingAvailables: HousingAvailable[]; // should be "any"?
+  page: any;
```
Comment (zaclittleberry) [inline src/app/_pages/housing-availables/housing-availables.page.ts:15]: no need for this. see below

File: src/app/_pages/housing-availables/housing-availables.page.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { Component, ViewChild } from '@angular/core';
+import { ApiService, HeadService } from '_services/index';
+import { HousingAvailable } from '_models/housing-available.model';
+import { Body } from '@angular/http/src/body';
```
Comment (zaclittleberry) [inline src/app/_pages/housing-availables/housing-availables.page.ts:4]: What is this for? I don't _think_ it's needed, but you can explain if it is.

**Review by zaclittleberry — CHANGES_REQUESTED** [review]

File: src/app/_pages/housing-availables/housing-availables.page.ts (hunk 3 above)
Comment (VitaRox) [reply src/app/_pages/housing-availables/housing-availables.page.ts:15]: See my reply below. :D

File: src/app/_pages/housing-availables/housing-availables.page.ts (hunk 4 above)
Comment (VitaRox) [reply src/app/_pages/housing-availables/housing-availables.page.ts:4]: This declares an empty array of type HousingAvailable, a type established and exported by '_models/housing-available.model'.  In our meeting you had instructed me to include this, but told me to type it as 'any', which i found didn't work; however, the moment I switched the array to have a declared type of HousingAvailable[], Typescript liked it and it ran fine for the first time (displaying the cards and their info).

**Pull request closed without merging**